cpp/CMakeLists.txt

```diff
@@ -505,10 +505,11 @@ add_library(cudftestutil STATIC
 
 target_compile_options(cudftestutil
             PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUDF_CXX_FLAGS}>"
-                   "$<$<COMPILE_LANGUAGE:CUDA>:${CUDF_CUDA_FLAGS}>"
+                   "$<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CUDA>:${CUDF_CUDA_FLAGS}>>"
 )
 
-target_compile_features(cudftestutil PUBLIC cxx_std_14 cuda_std_14)
+target_compile_features(cudftestutil
+    PUBLIC cxx_std_14 $<BUILD_INTERFACE:cuda_std_14>)
```

Comment:
Should the `cuda_std_14` use a `$<COMPILE_LANGUAGE:CUDA>` generator expression?

Reply:
That would be a safer approach compared to using BUILD_INTERFACE

Reply:
`$<COMPILE_LANGUAGE:CUDA>` isn't supported by `target_compile_features`, so this is the best approach.

Since `cxx_std_14` also implies `cuda_std_14` when `CUDA` is enabled, this shouldn't be a problem for downstream users